pkg/itemoperation/backup_operation.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package itemoperation
+
+// BackupOperation stores information about an async item operation
+// started by a BackupItemAction plugin (v2 or later)
+type BackupOperation struct {
+	Spec BackupOperationSpec `json:"spec"`
+
+	Status OperationStatus `json:"status"`
+}
+
+type BackupOperationSpec struct {
+	// BackupName is the name of the Velero backup this item operation
+	// is associated with.
+	BackupName string `json:"backupName"`
+
+	// BackupUID is the UID of the Velero backup this item operation
+	// is associated with.
+	BackupUID string `json:"backupUID"`
+
+	// BackupItemAction is the name of the BackupItemAction plugin that started the operation
+	BackupItemAction string `json:"backupItemAction"`
+
+	// Kubernetes resource identifier for the item
+	ResourceIdentifier string "json:resourceIdentifier"
+
+	// OperationID returned by the BIA plugin
+	OperationID string "json:operationID"
```

Comment:
I totally understand your concern. On the other hand, I don't know which way is more reasonable for our async operation model. Let's map this to a case in the real life:

- Person A asks Person B to do a business in a place
- Since the place is far away, B decides to rent a car to go there. So renting a car is purely an internal operation and the key of the car could be an internal info, a.k.a. the context
- B needs to return the car after the business is done but before he reports the completion to A

If we don't allow B to have any internal state and context, things have to be like this:

- Person A asks Person B to do a business in a place
- B has to ask Person C to do this because it cannot rent a car and keep the key
- A cannot communicate with C directly because C is the personal agent of B
- For anything, A has to ask B and B then asks C

Is this what we want? How do we define B's role as it looks very thin?

Reply:
There's nothing in the approved design that would account for augmenting the api with plugin-specific data structures in response. This won't be needed for any of the currently-identified use cases (datamover, etc.). SInce it's not in the approved design and adding it to the GRPC API might be challenging, I would suggest putting anything like that off for a future v3 API update if and when we identify a use case that requires it.

Reply:
This indeed affects one of the two proposals in the current Velero built-in data movement design, let's discuss this in the community meeting.

If we decide to drop the proposal and take the other one, I have no other concern for postposing this to v3 API, just one thing to mention:
**In the v2 API, it is not possible for the plugins to support restart-recovery.** It doesn't like that the implementation could decide whether and how it could support restart-recovery.
Still take the above A, B, C for example, because of the v2 API, B cannot keep internal state and context, so C has to do this. However, image how C does this, impossible -- In order to do this, C has to persist something or let other counterpart to keep something:

- If C persists something, it means we need to develop another mechanism something like the json, it is very complicated
- If C lets counterpart to keep something, the counterpart need to have the same lifecycle with the task(or in another words, the owner of the task), that is A. Unfortunately, A cannot communicate with C

Then going to the data mover case, once the module that hosts the SnapshotBackup/Restore controller restarts, the existing backup/restore will fail. This is the same as the existing PodVolume backup/restore, but because of the v2 API restriction, we cannot do anything better.

Reply:
OK, I think we have 2 questions to answer here:
1) do we need to add plugin-specific information to the interface as defined in the previously-approved designs
2) If yes to above, when is it needed.

Either way, if we need 1), someone will need to submit a design proposal for these additions. Since we don't even know for sure whether the internal datamover needs this, we're not in a position to implement and merge a PR based on this right now. This work can be done in follow-on PRs as modifications to the implementation of the previously-approved design.

As for 2) -- when we need this, if the internal datamover needs this, then we may want to get it into v2 API -- but the difference between getting it in v2 vs. v3 API is just whether we intend to merge it for Velero 1.11 vs. waiting for 1.12. Either way, it should be done as follow-on PRs since holding up current implementation of an approved design for some additional features that haven't been fully defined yet would endanger the release schedule. Lets implement the current design as agreed while we start iterating on what we want to add, if anything. Even if we decide we need it for 1.11, we're not going to save any time by holding current PRs while we iterate over the next design phase.

Reply:
Agree on the plan, let's see if the bult-in data movement needs it after the discussion.
And as we discussed in the meeting, if we need to add another change to the current BIA/RIA v2 interface, let's discuss and see if V3 is really required or we add the same to v2